lib/helpers/query/castUpdate.js

```diff
@@ -41,7 +40,11 @@ module.exports = function castUpdate(schema, obj, options, context, filter) {
     }
     return obj;
   }
-
+  if (schema.options.strict === 'throw') {
+    throw new StrictModeError(schema.options.discriminatorKey);
+  } else if (!context._mongooseOptions.hasOwnProperty('overwriteDiscriminatorKey') && context._mongooseOptions != null) {
```

Comment:
The `context._mongooseOptions != null` check should come first, because that's supposed to prevent `context._mongooseOptions.hasOwnProperty` from throwing an error.